src/runtime_src/core/common/utils.h

```diff
@@ -142,6 +147,20 @@ XRT_CORE_COMMON_EXPORT
 int
 get_pid();
 
+/**
+ * @brief Retrieve the last system error message.
+ *
+ * This function fetches the last error message from the operating system,
+ * which could originate from a Linux or Windows system call. It provides
+ * a human-readable string describing the error.
+ *
+ * @return A string containing the last system error message.
+ */
+inline std::string get_sys_last_err_msg()
```

Comment:
Let's make non-inline.  Move the definition to utils.cpp

```suggestion
XRT_CORE_COMMON_EXPORT
std::string 
get_sys_last_err_msg();
```